autoharness/orchestrator/orchestrator.py

```
@@ -0,0 +1,1077 @@
+
+        task_path = (Path(TASKS_DIR) / task_name).resolve()
+        tasks_root = Path(TASKS_DIR).resolve()
+        if not str(task_path).startswith(str(tasks_root)):
+            return _err({"error": "task_name escapes tasks directory"})
+        if not task_path.exists():
+            return _err({"error": f"Task '{task_name}' not found"}, 404)
+
+        timeout = inp.get("timeout", HARBOR_TIMEOUT)
+        start = time.time()
+        try:
+            result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)